src/components/app/App.js

```diff
@@ -32,8 +47,10 @@ class App extends Component {
         <h1>Rancid Tomatillos</h1>
       </header>
       <main>
-        {!this.state.selectedMovie && <Posters movies={this.state.movies} selectMovie={this.selectMovie}/>}
-
+        {this.state.errorCode && !this.state.fetchingMovie && <ErrorCode code={this.state.errorCode}/>}
+        {this.state.errorCode && this.state.fetchingMovie && <ErrorCode code={this.state.errorCode} fetchingMovie={this.state.fetchingMovie} clearSelected={this.clearSelected}/>}
+        {!this.state.errorCode && !this.state.movies.length && <Posters />}
+        {this.state.movies.length && !this.state.selectedMovie && <Posters movies={this.state.movies} selectMovie={this.selectMovie}/>}
         {this.state.selectedMovie && <MovieDetails movie={this.state.selectedMovie} clearSelected={this.clearSelected}/>}
```

Comment:
This is getting a little hairy here. Is there any way you could break these conditionals out of the render? Maybe perform the conditional checks above, and store the output in a variable? There's also a TON of `this.state` in here. Maybe destructuring would dry this up?